src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -216,6 +216,46 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             "transmute" => {
                 self.copy_op_transmute(args[0], dest)?;
             }
+            "ptr_offset_from" => {
+                let a = self.read_immediate(args[0])?.to_scalar()?.to_ptr()?;
+                let b = self.read_immediate(args[1])?.to_scalar()?.to_ptr()?;
+                if a.alloc_id != b.alloc_id {
+                    throw_ub_format!(
+                        "ptr_offset_from cannot compute offset of pointers into different \
+                        allocations.",
+                    );
+                }
+                let ptr_size = self.tcx.data_layout.pointer_size;
+                let a = Scalar::from_uint(a.offset.bytes(), ptr_size);
+                let b = Scalar::from_uint(b.offset.bytes(), ptr_size);
+                let usize_layout = self.layout_of(self.tcx.types.usize)?;
+                let a = ImmTy {
```

Comment:
Please use `ImmTy::from_scalar` instead of calling the struct constructor.

Reply:
But, in fact, you can use `ImmTy::from_uint` (you marked that comment as resolved but did not actually implement it?)

Reply:
I marked it as done as I did it locally but I never pushed, because I was running into other problems